src/realm/sync/client.cpp

```diff
@@ -1783,6 +1784,12 @@ void Client::stop() noexcept
 }
 
 
+void Client::post_for_testing(SyncSocketProvider::FunctionHandler&& handler)
```

Comment:
One thing I just realized is that none of the new post functions (i.e, that use FunctionHandler as parameter) have the argument (Status) needed by the handlers when they are executed. I guess Service class does not complain because it's templated (and accepts pretty much everything) and it probably assigns a default Status for the handlers.

Reply:
It was actually updated _network.hpp_ in PR #6116 to provide a Status back to the handler:
https://github.com/realm/realm-core/blob/master/src/realm/sync/network/network.hpp#L2018
Timers were updated as well:
https://github.com/realm/realm-core/blob/master/src/realm/sync/network/network.hpp#L3510

Reply:
Got it. But shouldn't the caller provide such a status? Or maybe I completely misunderstood its usage.

Reply:
The status value is meant to be provided by the event loop based on what happened while the handler was in the event loop queue and transitioning to executing. For now `post()` doesn't have a status other than OK, since posted handlers can't be canceled. Timer's on the other hand, can provide either OK or Aborted depending on whether or not the timer was canceled.

Reply:
I see. I thought websockets can report a status too since it's ErrorCodes we use.

Reply:
They will need to in the future... that's a later PR - for now, we're still using the old mechanism.